### ISSUE TITLE
mitmproxy handles client certificate incorrectly

### ISSUE THREAD

**Issue by stashme** [mitmproxy/mitmproxy#3937]
#### Problem Description
For the scenario in question, mitmproxy needs to authenticate against the server using a client certificate. The server itself trusts the root ca only, the client certificate was issued by an intermediate ca. 
So the chain looks like this: client cert --> intermediate ca --> root ca
```
openssl s_client -connect example.com:443 -prexit
Acceptable client certificate CA names
CN = Root CA
````
config.yaml contains "client_certs: "/home/mitmproxy/private/cert.pem"
cert.pem looks like this:
```
-----BEGIN RSA PRIVATE KEY-----
(Private Key)
-----END RSA PRIVATE KEY-----
-----BEGIN CERTIFICATE-----
(Client Cert)
-----END CERTIFICATE-----
-----BEGIN CERTIFICATE-----
(Intermediate CA)
-----END CERTIFICATE----- 
```
I successfully connected to the server using curl, the client cert was accepted.

However, mitmproxy ignores the intermediate CA and sends only the client cert in the TLS message. I used tcpdump to dump the connection with two different scenarios: with curl, as well as with mitmproxy connecting to the server.
It is clearly visible in the TLS handshake (TLSv1.2 Certificate, Client Key Exchange, Certificate Verify, Change Cipher Spec, Encrypted Handshake Message) that there is only **one** client certificate sent with mitmproxy, but **two** with curl. The intermediate ca is missing with mitmproxy, thus the server rejects the connection ("sslv3 alert certificate unknown")

I also checked [stackexchange.com](https://security.stackexchange.com/questions/206468/does-client-have-to-send-the-ca-chain-along-with-the-client-certificate-after-se), it seems curl works as expected and my pem file is ok.

#### Steps to reproduce the behavior:
1. Have an SSL server that accepts client certificate signed by a Root CA
2. Have an intermediate CA that is signed by the Root CA
3. Have a client cert that is signed by the intermediate CA
4. Prepare a pem file with private key, client cert and intermediate ca
5. Use mitmproxy with "client_certs: "cert.pem"

#### System Information
Mitmproxy: 5.0.1
Python:    3.6.9
OpenSSL:   OpenSSL 1.0.2u  20 Dec 2019
Platform:  Linux-4.15.0-91-generic-x86_64-with

**Comment by mhils** [comment]
Thanks for the detailed report. This might just be us calling OpenSSL's `SSL_CTX_use_certificate_file` instead of `SSL_CTX_use_certificate_chain_file`:

https://github.com/mitmproxy/mitmproxy/blob/bfb8da4b1c5f1422b4a9031d1e3e77d40552208c/mitmproxy/net/tls.py#L297-L302

Can you try to replace `use_certificate_file` with `use_certificate_chain_file` and let us know if that fixes it?

**Comment by stashme** [comment]
That was exactly what was missing, works now as expected. Thanks a lot for your fast response! Best regards

**Comment by mhils** [comment]
Fixed in 7c72776, thanks!
